frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/ParametersSettings.tsx

```diff
@@ -0,0 +1,203 @@
+                      }
+                    >
+                      <Option
+                        icon="close"
+                        value="disabled"
+                      >{t`Disabled`}</Option>
+                      <Option
+                        icon="pencil"
+                        value="enabled"
+                      >{t`Editable`}</Option>
+                      <Option icon="lock" value="locked">{t`Locked`}</Option>
+                    </Select>
+                  </div>
+                ))}
+              </Stack>
+            </StaticEmbedSetupPaneSettingsContentSection>
+
+            {previewParameters.length > 0 && (
+              <>
+                <Divider my="2rem" />
+                <StaticEmbedSetupPaneSettingsContentSection
+                  title={t`Preview locked parameters`}
+                >
+                  <Stack spacing="1rem">
+                    <Text>{t`Try passing some sample values to your locked parameters here. Your server will have to provide the actual values in the signed token when doing this for real.`}</Text>
+
+                    {valuePopulatedParameters.map(parameter => (
+                      <StaticParameterWidget
+                        key={parameter.id}
+                        className="m0"
+                        parameter={parameter}
+                        parameters={valuePopulatedParameters}
+                        setValue={(value: string) =>
+                          onChangeParameterValue(parameter.id, value)
+                        }
+                      />
+                    ))}
+                  </Stack>
+                </StaticEmbedSetupPaneSettingsContentSection>
+              </>
+            )}
+          </>
+        ) : (
+          <>
+            <Box mb="2rem">
+              {t`This ${resourceType} doesn't have any parameters to configure yet.`}
+            </Box>
+            <Divider />
+          </>
+        )
+      }
+      previewSlot={
+        <>
+          <SegmentedControl
+            value={activePane}
+            data={CODE_PREVIEW_CONTROL_OPTIONS}
+            onChange={onChangePane}
+          />
+
+          {activePane === "preview" ? (
+            <PreviewPane
+              className="flex-full"
+              previewUrl={iframeUrl}
+              isTransparent={displayOptions.theme === "transparent"}
+            />
+          ) : activePane === "code" ? (
+            <EmbedCodePane
+              className="flex-full w-full"
+              embedType={embedType}
+              resource={resource}
+              resourceType={resourceType}
+              iframeUrl={iframeUrl}
+              token={token}
+              siteUrl={siteUrl}
+              secretKey={secretKey}
+              params={params}
+              displayOptions={displayOptions}
+            />
+          ) : null}
+        </>
+      }
+    />
+  );
+};
+
+const getIconForParameter = (parameter: EmbedResourceParameter) =>
```

Comment:
I looked at this for 20 seconds and my brain hurts, maybe we can simplify this one a little?